iit/model_pairs/strict_iit_model_pair.py

```diff
@@ -21,14 +21,9 @@ def __init__(
             training_args: dict = {}
             ):
         default_training_args = {
-            "batch_size": 256,
-            "lr": 0.001,
-            "num_workers": 0,
-            "use_single_loss": False,
-            "iit_weight": 1.0,
-            "behavior_weight": 1.0,
             "strict_weight": 1.0,
-            "clip_grad_norm": 1.0,
+            "strict_weight_schedule" : lambda s, i: s,
```

Comment:
This is a cool idea! 

Maybe it is better to implement it as 

```
@property 
def strict_weight_at_epoch(self):
    return self.training_args.strict_weight_schedule(<args_from_self>)
```

Instead of changing the strict weight variable after each epoch? (or a method like `strict_weight_for_epoch = self.get_scheduled_strict_weight()` and then calculate the loss). 

This lambda is also throwing me off a bit, maybe renaming the args will make it clearer...

It also seems like this is achievable by using different optimisers/lrs for each loss (and not using single loss)? No idea which one's better though...

Reply:
Hm, unclear to me which is the right way to go right now. I think it's best to remove it (right now it's not doing anything) and if you find that this is a useful idea down the road you can add it how you see fit?